src/main/java/org/jnode/fs/xfs/Superblock.java

```diff
@@ -156,177 +193,103 @@ public Superblock(XfsFileSystem fileSystem) throws FileSystemException {
+
+        realTimeExtentSize = readUInt32(); // sb_rextsize
+        agBlockSize = readUInt32(); // sb_agblocks
+        agCount = readUInt32(); // sb_agcount
+        realTimeBitmapBlockCount = readUInt32(); // sb_rbmblocks
+        logBlockCount = readUInt32(); // sb_logblocks
+
+        versionNumer = readUInt16(); // sb_versionnum, a bitmask of VersionFlags
+        sectorSize = readUInt16(); // sb_sectsize
+        inodeSize = readUInt16(); // sb_inodesize
+        inodePerBlock = readUInt16(); // sb_inopblock
+
+        int nameLength = 12;
+        byte[] buffer = new byte[nameLength];
+        System.arraycopy(getData(), getOffset(), buffer, 0, nameLength);
+        fileSystemName = new String(buffer, StandardCharsets.UTF_8).replace("\0", ""); // sb_fname[12]
+        skipBytes(nameLength);
+
+        blockSizeLog2 = readUInt8(); // sb_blocklog
+        sectorSizeLog2 = readUInt8(); // sb_sectlog
+        inodeSizeLog2 = readUInt8(); // sb_inodelog
+        iNodePerBlockLog2 = readUInt8(); // sb_inopblog
+        agSizeLog2 = readUInt8(); // sb_agblklog
+        realTimeExtentSizeLog2 = readUInt8(); // sb_rextslog
+        inProgressFlag = readUInt8(); // sb_inprogress
+        maxInodePercentage = readUInt8(); // sb_imax_pct
+
+        allocatedInodeCount = readInt64(); // sb_icount
+        freeInodeCount = readInt64(); // sb_ifree
+        freeDataBlockCount = readInt64(); // sb_fdblocks
+        freeRealTimeExtentCount = readInt64(); // sb_frextents
+        userQuotaInode = readInt64(); // sb_uquotino
+        groupQuotaInode = readInt64(); // sb_gquotino
+
+        quotaFlags = readUInt16(); // sb_qflags
+        flags = readUInt8(); // sb_flags
+        sharedVersionNumber = readUInt8(); // sb_shared_vn
+        inodeChunkAlignment = readUInt32(); // sb_inoalignmt
+
+        unit = readUInt32(); // sb_unit
+        width = readUInt32(); // sb_width
+
+        directoryBlockLog2 = readUInt8(); // sb_dirblklog
+        externalLogSectorSizeLog2 = readUInt8(); // sb_logsectlog
+        externalLogSectorSize = readUInt16(); // sb_logsectsize
+        logUnitSize = readUInt32(); // sb_logsunit
+
+        additionalFeatureFlags = readUInt32(); // sb_features2, the bitmask of Features2
+        additionalFeatureFlagsMirror = readUInt32(); // sb_bad_features2
+
+        // version 5 superblock fields start here
+        readWriteCompatibleFlags = readUInt32(); // sb_features_compat
+        readOnlyCompatibleFlags = readUInt32(); // sb_features_ro_compat
+        readWriteIncompatibleFlags = readUInt32(); // sb_features_incompat
+        readWriteLogcompatibleFlags = readUInt32(); // sb_features_log_incompat
+        superblockChecksum = readUInt32(); // sb_crc
+        sparseInodeAlignment = readUInt32(); // sb_spino_align
+
+        projectQuotaInode = readInt64(); // sb_pquotino
+        logSequenceNumber= readInt64(); // sb_lsn , TODO, there is not definition of xfs_lsn_t in the document..
```

Comment:
![image](https://user-images.githubusercontent.com/82918854/192967172-fa99df23-e630-40a2-8bd9-aa68adb7b0a8.png)

Reply:
But there is no definition of xfs_lsn_t..

Reply:
One reference here https://elixir.bootlin.com/linux/v3.12.47/source/fs/xfs/xfs_types.h#L35
Unfortunately not in the XFS PDF docs or https://github.com/libyal/libfsxfs/blob/main/documentation/X%20File%20System%20(XFS).asciidoc 
but maybe better yet? ... https://github.com/torvalds/linux/blob/master/fs/xfs/libxfs/xfs_types.h#L23

Reply:
Yeah, In Torvalds, We Trust. haha